barretenberg_wasm/src/lib.rs

```diff
@@ -235,32 +178,84 @@ impl Barretenberg {
         Barretenberg { memory, instance }
     }
 }
-#[allow(unused_variables)]
-fn logstr(ptr: i32) {
-    // println!("[No logs]")
-    // let memory = my_env.memory.get_ref().unwrap();
-
-    // let mut ptr_end = 0;
-    // let byte_view = memory.uint8view();
-
-    // for (i, cell) in byte_view[ptr as usize..].iter().enumerate() {
-    //     if cell != 0 {
-    //         ptr_end = i;
-    //     } else {
-    //         break;
-    //     }
-    // }
-
-    // let str_vec: Vec<_> = byte_view[ptr as usize..=(ptr + ptr_end as i32) as usize]
-    //     .into_iter()
-    //     .cloned()
-    //     .collect();
-
-    // // Convert the subslice to a `&str`.
-    // let string = std::str::from_utf8(&str_vec).unwrap();
-
-    // // Print it!
-    // println!("[WASM LOG] {}", string);
+
+fn logstr(env: &Env, ptr: i32) {
+    let mut ptr_end = 0;
+    let byte_view = env.memory.uint8view();
+
+    for (i, cell) in byte_view[ptr as usize..].iter().enumerate() {
+        if cell != &Cell::new(0) {
+            ptr_end = i;
+        } else {
+            break;
+        }
+    }
+
+    let str_vec: Vec<_> = byte_view[ptr as usize..=(ptr + ptr_end as i32) as usize]
+        .iter()
+        .cloned()
+        .map(|chr| chr.get())
+        .collect();
+
+    // Convert the subslice to a `&str`.
+    let string = std::str::from_utf8(&str_vec).unwrap();
+
+    // Print it!
+    println!("{string}");
+}
+
+// TODO: This should actually be creating random data, right?
+// Probably duplicate https://github.com/wasmerio/wasmer/blob/2.3.0/lib/wasi/src/syscalls/mod.rs#L2537
+fn random_get(_: i32, _: i32) -> i32 {
+    0_i32
+}
```

Comment:
@kevaundray what should be the priority on fixing this? Through some logging, it seems that random_get is actually getting called quite a bit

Reply:
I think we can merge without it with a tracking issue if this is going to take too much time to fix

Reply:
```suggestion
// TODO: Implement randomization function https://github.com/noir-lang/aztec_backend/issues/101
```

Now we have the issue, let's add a link to it.

Reply:
I might just finish it before this lands 🙃